cadence/contracts/ISPOManager.cdc

```diff
@@ -625,25 +666,42 @@ pub contract ISPOManager {
             return ISPOClientInfo(
                 ispoId: self.ispoId,
                 delegatedFlowBalance: self.getDelegatedFlowBalance(),
-                rewardTokenBalance: self.getRewardTokenBalance(epoch: FlowEpochProxy.getCurrentEpoch()),
+                // - 1 epoch because the rewards for the current epoch are not yet finalized
+                rewardTokenBalance: self.getRewardTokenBalance(),
                 createdAt: self.createdAt,
             )
         }
 
         pub fun hasDelegation(): Bool {
             return ISPOManager.borrowISPORecord(id: self.ispoId).hasNodeDelegator(delegatorId: self.uuid)
         }
+
         // TODO: destroy
     }
 
     pub fun createISPOClient(ispoId: UInt64, flowVault: @FungibleToken.Vault): @ISPOClient {
       return <- create ISPOClient(ispoId: ispoId, flowVault: <- flowVault, createdAt: getCurrentBlock().timestamp)
     }
 
+    access(self) fun min(a: UInt64, b: UInt64): UInt64 {
+        if a < b {
+            return a
+        }
+        return b
+    }
+
+    access(self) fun max(a: UInt64, b: UInt64): UInt64 {
+        if a > b {
+            return a
+        }
+        return b
+    }
+
     init() {
         self.ispos <- {}
-        self.defaultNodeId = ""
+        self.defaultNodeId = "26c1cd3254ec259b4faea0f53e3a446539256d81f0c06fff430690433d69731f"
```

Comment:
I'd drop defaultNodeId completely, it unnecessarily complicates the deployment of the contract across networks and we don't use that feature

Reply:
or maybe just make it `nil` and keep it there just in case